crates/sc-proof-of-time/src/verifier.rs

```diff
@@ -298,7 +298,7 @@ impl PotVerifier {
     ) -> bool {
         // TODO: This "proxy" is a workaround for https://github.com/rust-lang/rust/issues/57478
         let (result_sender, result_receiver) = oneshot::channel();
-        rayon::spawn({
+        tokio::task::spawn_blocking({
```

Comment:
Looks like the ugliness in L305-315 can be avoided if `verify_checkpoints_internal()` can be made a stand alone fn that takes `self.cache.clone()` as input?

Reply:
The issue is the lock held in async context, not the fact that we use `self` or where `cache` comes from